ServicePackaging.md

```diff
@@ -0,0 +1,533 @@
+}
+EOT
+elif [[ $home_type = "filebrowser" ]]; then
+    directory=$(yq e '.homepage.directory' start9/config.yaml)
+    cat >> /etc/nginx/conf.d/default.conf <<EOT
+server {
+  autoindex on;
+  listen 80;
+  listen [::]:80;
+  server_name ${TOR_ADDRESS};
+  root "/root/start9/public/filebrowser/${directory}";
+}
+EOT
+else
+    cat >> /etc/nginx/conf.d/default.conf <<EOT
+server {
+  listen 80;
+  listen [::]:80;
+  server_name ${TOR_ADDRESS};
+  root "/var/www/${home_type}";
+}
+EOT
+fi
+
+for subdomain in "${subdomains[@]}"; do
+    subdomain_type=$(yq e ".subdomains.[] | select(.name == \"$subdomain\") | .settings |.type" start9/config.yaml)
+    if [[ $subdomain_type == "filebrowser" ]]; then
+        directory="$(yq e ".subdomains.[] | select(.name == \"$subdomain\") | .settings | .directory" start9/config.yaml)"
+        cat >> /etc/nginx/conf.d/default.conf <<EOT
+server {
+  autoindex on;
+  listen 80;
+  listen [::]:80;
+  server_name ${subdomain}.${TOR_ADDRESS};
+  root "/root/start9/public/filebrowser/${directory}";
+}
+EOT
+    elif [ $subdomain_type = "redirect" ]; then
+        if [ "$(yq e ".subdomains.[] | select(.name == \"$subdomain\") | .settings | .target == ~" start9/config.yaml)" = "true"]; then
+            cat >> /etc/nginx/conf.d/default.conf <<EOT
+server {
+  listen 80;
+  listen [::]:80;
+  server_name ${subdomain}.${TOR_ADDRESS};
+  return 301 http://${TOR_ADDRESS}$request_uri;
+}
+EOT
+        else
+            target="$(yq e ".subdomains.[] | select(.name == \"$subdomain\") | .settings | .target" start9/config.yaml)"
+            cat >> /etc/nginx/conf.d/default.conf <<EOT
+server {
+  listen 80;
+  listen [::]:80;
+  server_name ${subdomain}.${TOR_ADDRESS};
+  return 301 http://${target}.${TOR_ADDRESS}$request_uri;
+}
+EOT
+        fi
+    fi
+done
+
+exec tini -- nginx -g "daemon off;"
+```
+
+Additional details on the `Dockerfile` and `entrypoint` can be found [here](https://docs.start9.com/contributing/services/docker.html).
+
+### Makefile (Optional)
+
+Here, we will create a `Makefile`, which is optional, but recommended as it outlines the build and streamlines additional developer contributions.  Alternatively, you could use the `nix` specification instead of `make`.
+
+Our example `Makefile` is fairly simple for Embassy Pages.  Let's take a look:
+
+```
+.DELETE_ON_ERROR:
+
+all: embassy-pages.s9pk
+
+install: embassy-pages.s9pk
+	appmgr install embassy-pages.s9pk
+
+embassy-pages.s9pk: manifest.toml assets/compat/config_spec.yaml image.tar instructions.md
+	embassy-sdk pack
+
+image.tar: Dockerfile docker_entrypoint.sh
+	DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --tag start9/embassy-pages/main:0.1.3 --platform=linux/arm/v7 -o type=docker,dest=image.tar .
+```
+
+1. The first line simply removes the progress of a `make` process if it fails.
+`.DELETE_ON_ERROR:`
+
+2. The `all` step is run when the `make` command is issued.  This means that the `embassy-pages.s9pk` must first be built, and so on.
```

Comment:
The Makefile format has changed substantially in 030, see [lnd-wrapper for 0.3.0](https://github.com/Start9Labs/lnd-wrapper/pull/24/files)

Reply:
I think will move to hello-world, and my next project probably ought to be to get that updated for 0.3.0